### PR TITLE
Create end to end integration test for DevTools

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This PR adds the first end to end integration test for DevTools, and enables the integration tests to run on the CI. The test app_test.dart is very simple - it launches devtools, verifies we can connect to a real flutter app, and then navigates to every screen.

RELEASE_NOTE_EXCEPTION=testing change only

### CODE REVIEW

File: packages/devtools_app/integration_test/test/app_test.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  });
+
+  testWidgets('connect to app and switch tabs', (tester) async {
+    await pumpDevTools(tester);
+    expect(find.byType(LandingScreenBody), findsOneWidget);
+    expect(find.text('No client connection'), findsOneWidget);
+
+    // temporary, just committed this to verify that the bots will fail if this
+    // test case fails.
+    expect(true, isFalse);
```
Comment (elliette) [inline packages/devtools_app/integration_test/test/app_test.dart:33]: Commenting as a reminder to remove :)

File: packages/devtools_app/integration_test/test/test_utils.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    if (serviceUri == null) {
+      throw Exception('Cannot create a TestApp with a null service uri.');
+    }
+    return TestApp._(vmServiceUri: serviceUri);
+  }
+
+  factory TestApp.fromEnvironment() {
+    const testArgs = String.fromEnvironment('test_args');
+    final Map<String, Object> argsMap =
+        (jsonDecode(testArgs)).cast<String, Object>();
```
Comment (elliette) [inline packages/devtools_app/integration_test/test/test_utils.dart:63]: are the `( )` around `jsonDecode` necessary?

File: packages/devtools_app/integration_test/test_infra/test_app_driver.dart [hunk 3]
```diff
@@ -106,34 +109,78 @@ abstract class _TestApp with IOMixin {
     _vmServiceWsUri =
         convertToWebSocketUrl(serviceProtocolUrl: _vmServiceWsUri);
 
-    await started;
+    // Now await the started event; if it had already happened the future will
+    // have already completed.
+    final startedResult = await started;
+    _currentRunningAppId = (startedResult[FlutterDaemonConstants.params.key]!
```
Comment (elliette) [inline packages/devtools_app/integration_test/test_infra/test_app_driver.dart:115]: Consider creating variables for the `values` being casted to make it easier to read

**Review by elliette — APPROVED** [review]

File: packages/devtools_app/integration_test/test/app_test.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools_app/integration_test/test/app_test.dart:33]: TODO to myself before landing - make sure the test fails on the CI when the integration test fails.

File: packages/devtools_app/integration_test/test_infra/_run_test.dart [hunk 4]
```diff
@@ -103,7 +107,12 @@ class TestRunner with IOMixin {
         if (enableExperiments) '--dart-define=enable_experiments=true',
       ],
     );
-    listenToProcessOutput(process);
+    listenToProcessOutput(
+      process,
+      onError: (line) {
+        print('on error, would be throwing exception here: $line');
```
Comment (jacob314) [inline packages/devtools_app/integration_test/test_infra/_run_test.dart:113]: nit: I'd make this more concise, e.g.
`$processName: $line` or
`chromeDriver: $line`

Comment (kenzieschmoll) [reply packages/devtools_app/integration_test/test_infra/_run_test.dart:113]: this print was just for debugging, but will apply this comment if we end up throwing an exception from this callback 👍

**Pull request merged**